### PR TITLE
feat(dispatcher): add web identity support to the AWS KMS signer

### LINKED ISSUE #6: AWS KMS signer for the Dispatcher
## Context

Currently the dispatcher signs transactions by instantiating a local wallet.
It creates the wallet through a mnemonic read from a ENV variable or the configuration file.
This is not secure at all.

## Solution

It should be possible to use a signing API like AWS KMS.

### Subtasks

- [x] Change the transaction manager configuration.
- [x] Add the proper configurations to the dispatcher.
- [x] Add support for conditional signer instantiation in the dispatcher.
- [x] Implement the AWS KMS signer.

### PR DESCRIPTION
**Pull request by renan061**

### CODE REVIEW

File: offchain/dispatcher/src/signer/aws_credentials.rs [hunk 1]
```diff
@@ -0,0 +1,157 @@
+use async_trait::async_trait;
```
Comment (gligneul) [inline offchain/dispatcher/src/signer/aws_credentials.rs:1]: Missing copyright

File: offchain/dispatcher/src/signer/aws_credentials.rs [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+use rusoto_core::credential::{
+    AwsCredentials, CredentialsError, DefaultCredentialsProvider,
+    ProvideAwsCredentials,
+};
+use rusoto_sts::WebIdentityProvider;
+use std::env;
+
```
Comment (gligneul) [inline offchain/dispatcher/src/signer/aws_credentials.rs:8]: Could you add an explanation for this module?

File: offchain/dispatcher/src/signer/aws_credentials.rs [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+        for env_var in WEB_IDENTITY_ENV_VARS {
+            if env::var(env_var).is_err() {
+                tracing::trace!("ENV VAR {} is not set", env_var);
+                tracing::trace!("instantiating default provider");
+                return DefaultCredentialsProvider::new()
+                    .map(Box::new)
+                    .map(|inner| Self(inner));
+            }
+        }
+        tracing::trace!("instantiating web identity provider",);
```
Comment (gligneul) [inline offchain/dispatcher/src/signer/aws_credentials.rs:36]:
```suggestion
        tracing::trace!("instantiating web identity provider");
```

File: offchain/dispatcher/src/signer/aws_signer.rs [hunk 4]
```diff
@@ -149,3 +124,17 @@ impl Signer for AwsSigner {
...
+pub mod tests {
+    use rusoto_core::Region;
+
+    use crate::signer::aws_signer::AwsSigner;
+
+    #[tokio::test]
+    async fn new_aws_signer_with_error() {
+        let aws_signer =
+            AwsSigner::new("invalid".to_string(), 0, Region::UsEast1).await;
+        assert!(aws_signer.is_err());
```
Comment (gligneul) [inline offchain/dispatcher/src/signer/aws_signer.rs:138]: The error happening here is because the key is invalid?

Comment (renan061) [reply offchain/dispatcher/src/signer/aws_signer.rs:138]: Correct! Added a commentary.

**Review by gligneul — APPROVED** [review]

**Pull request merged**